diff --git a/node_modules/react-native-drawer-layout/src/views/modern/Drawer.tsx b/node_modules/react-native-drawer-layout/src/views/modern/Drawer.tsx
index 9393ef6..ec48ae4 100644
--- a/node_modules/react-native-drawer-layout/src/views/modern/Drawer.tsx
+++ b/node_modules/react-native-drawer-layout/src/views/modern/Drawer.tsx
@@ -317,6 +317,7 @@ export function Drawer({
   });
 
   const drawerAnimatedStyle = useAnimatedStyle(() => {
+    const distanceFromEdge = layout.width - drawerWidth;
     return {
       transform:
         drawerType === 'permanent'
@@ -327,7 +328,8 @@ export function Drawer({
               {
                 translateX:
                   // The drawer stays in place when `drawerType` is `back`
-                  drawerType === 'back' ? 0 : translateX.value,
+                    (drawerType === 'back' ? 0 : translateX.value) +
+                    (drawerPosition === 'left' ? 0 : distanceFromEdge),
               },
             ],
     };
